[vivienherq] iP

vivienherq opened the pull request:
### Dude

Dude frees your mind of having to remember things you need to do. It's:
- text-based
- easy to learn
- FAST SUPER FAST to use

All you need to do is:
1. download it from [here](https://github.com/vivienherq/ip).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:
- [ ]  Managing tasks
- [ ]  Managing deadlines 
- [ ]  Managing events
- [ ]  Reminders (coming soon)

---

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

vivienherq committed:
rename

vivienherq committed:
greet & exit

vivienherq committed:
echo

vivienherq committed:
store & list

vivienherq committed:
task class

vivienherq committed:
mark & unmark

vivienherq committed:
lvl4 todo, event & deadline classes

vivienherq committed:
lvl4 todo

vivienherq committed:
lvl4 create todos, events, deadlines

vivienherq committed:
lvl 5 handle errors

vivienherq committed:
lvl 6 delete

garylow2001 commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,16 @@
+        this.from = from;
+        this.to = to;
+    }
+    public String getType() {
```
Perhaps you can add a ```@Override``` above your ```getType()``` method to specify that you wish to override this method from Task

garylow2001 commented on src/main/java/Dude.java:
```diff
@@ -0,0 +1,122 @@
+
+//    static Task[] taskList = new Task[100];
+    static ArrayList<Task> taskList = new ArrayList<Task>();
+    static int nTasks = 0;
```
Instead of having an ```nTasks``` to keep track of the number of tasks in ```taskList``` why not use ```taskList.size()```?

garylow2001 commented on src/main/java/Dude.java:
```diff
@@ -0,0 +1,122 @@
+    public static void list() {
+        for (int i = 0; i < nTasks; i++) {
+            Task task = taskList.get(i);
+//            Task task = taskList[i];
```
You are using comments to temporarily remove code, however comments should be used to describe the code to the reader. They should also be indented to the relative position to your code i.e. the ```//``` needs to be on the same line as ```Task``` on line 47

garylow2001 commented on src/main/java/Dude.java:
```diff
@@ -0,0 +1,122 @@
+
+    public static void mark(int n) {
+//        taskList[n-1].setDone(true);
+        taskList.get(n-1).setDone(true);
```
For clarity, perhaps you can split this line up into
```
selectedTask = taskList.get(n-1);
selectedTask.setDone(true);
```

garylow2001 commented on src/main/java/Dude.java:
```diff
@@ -0,0 +1,122 @@
+            String input = scanner.nextLine();
+            String[] words = input.split(" ");
+
+            if (words[0].equals("list")) {
```
Perhaps you can use ```switch``` statements for the control flow instead of if-else statements

garylow2001 reviewed (commented):
LGTM. Overall just some nits about coding standards!

vivienherq committed:
add text ui testing

vivienherq committed:
add save tasks and load task

vivienherq committed:
implement datetime recognition for deadline

samuelim01 commented on src/main/java/Dude.java:
```diff
@@ -0,0 +1,213 @@
+import java.io.*;
```
Imported classes should be listed explicitly 😁 
```suggestion
import java.io.BufferedReader;
import java.io.File;
import java.io.FileNotFoundException;
import java.io.FileReader;
import java.io.FileWriter;
import java.io.IOException;
```

samuelim01 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,35 @@
+public class Task {
+    private String description;
+    private boolean done;
```
booleans variables should be named to sound like booleans
```suggestion
    private boolean isDone;
```

samuelim01 commented on src/main/java/Dude.java:
```diff
@@ -0,0 +1,213 @@
+    }
+
+    public static void unmark(int n) throws IOException {
+        taskList.get(n-1).setDone(false);
```
Operators should be surrounded by a space character.
```suggestion
        taskList.get(n - 1).setDone(false);
```

samuelim01 commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,33 @@
+import java.util.Locale;
+
+public class Deadline extends Task {
+    LocalDateTime by;
```
Variables should be non-public.
```suggestion
    private LocalDateTime by;
```

samuelim01 commented on src/main/java/Dude.java:
```diff
@@ -0,0 +1,213 @@
+    static ArrayList<Task> taskList = new ArrayList<Task>();
+    static int nTasks = 0;
+
+    private static final String FILE_PATH = "./data/dude.txt";
```
This constant doesn't seem to be used. Did you forget to replace your instances of `"./data/dude.txt"` with `FILE_PATH`?

samuelim01 commented on data/dude.txt:
This file can be excluded from commits!

samuelim01 reviewed (commented):
LGTM, just a few nits about coding standards. Overall, good job 😄

vivienherq committed:
Refactor code into Ui, Storage, Parser, TaskList, Command classes

vivienherq committed:
Refactor code into Ui, Storage, Parser, TaskList, Command classes

vivienherq committed:
Revert latest commit to push to A-MoreOOP Branch

vivienherq committed:
Organise classes into packages

vivienherq committed:
Merge branch 'A-Packages'

vivienherq committed:
Merge pull request #1 from vivienherq/add-gradle-support

Add Gradle support

vivienherq committed:
Add Gradle support

vivienherq committed:
Add JUnit tests

vivienherq committed:
Merge pull request #2 from vivienherq/A-JUnit

Add JUnit tests

vivienherq committed:
Modify build.gradle for JAR creation

andrefoo commented on src/main/java/dude/Ui.java:
```diff
@@ -0,0 +1,67 @@
+
+import java.util.Scanner;
+public class Ui {
+    Scanner sc;
```
Logical units within a block should be separated by one blank line.
Consider making the variable 'sc' private as well!

andrefoo reviewed (commented):
LGTM, minor fixes only.

vivienherq committed:
Implement find feature to search for tasks

vivienherq committed:
Add JavaDocs to methods and classes

vivienherq committed:
Modify code to apply coding standards

vivienherq committed:
Merge pull request #3 from vivienherq/branch-Level-9

Implement find feature to search for tasks

vivienherq committed:
Merge pull request #4 from vivienherq/branch-A-CodingStandard

Modify code to apply coding standards

vivienherq committed:
Merge branch 'master' into branch-A-JavaDoc

vivienherq committed:
Merge pull request #5 from vivienherq/branch-A-JavaDoc

Add JavaDocs to methods and classes

vivienherq committed:
Fix coding style violations

vivienherq committed:
Create GUI for Dude

vivienherq committed:
Add functionality to GUI for dude

vivienherq committed:
Add user interaction functionality to GUI for Dude

vivienherq committed:
Merge pull request #6 from vivienherq/branch-Level-10

Add GUI for Dude

vivienherq committed:
Add assertions

vivienherq committed:
Improve code quality

Improve code quality

Code quality is not up to standard.

Improvements are required to make the codebase more maintainable,
less error-prone and easier to work with.

* Refactored complex methods to improve readability and maintainability.
* Applied coding standards.
* Fixed checkstyle violations.
* Added JavaDocs to new added methods and classes.

vivienherq committed:
Merge pull request #7 from vivienherq/branch-A-Assertions

Add assertions

vivienherq committed:
Merge pull request #8 from vivienherq/branch-A-CodeQuality

Improve code quality

vivienherq committed:
Set up CI

Set up CI

To run workflow (as per gradle.ymp file) every time certain
project events are triggered.

vivienherq committed:
Merge pull request #9 from vivienherq/branch-A-CI

Set up CI

vivienherq committed:
Add Note and NoteList classes

vivienherq committed:
Add and modify methods to support Notes

vivienherq committed:
Fix Checkstyle violations

vivienherq committed:
Merge pull request #10 from vivienherq/branch-D-Notes

Add classes and methods to support Notes in Dude

vivienherq committed:
Reorganise project structure and add packages

Current structure is not very organised after adding new classes.

New packages are added and classes are moved into packages.

* dude.ui package added
* dude.note package added

vivienherq committed:
Fix Checkstyle violations

vivienherq committed:
Storage: prevent null values from being returned

Null values are returned for Task and Note.

Null values in loaded data can lead to unexpected behaviour
and errors in the application.

vivienherq committed:
Find command: make matching case-insensitive

Find command is case-sensitive.

A case-insensitive find is more user-friendly because users cannot be
expected to remember the exact case of the keywords.

* Update search algorithm to use case-insensitive matching

vivienherq committed:
Configure dimension and layout of GUI

vivienherq committed:
Improve GUI

vivienherq committed:
Merge pull request #12 from vivienherq/branch-A-BetterGui

Branch a better gui

vivienherq committed:
Add delete note command

vivienherq committed:
Add additional checks for error handling

vivienherq committed:
Add exceptions to Parser class

vivienherq committed:
Add exceptions to Storage class

vivienherq committed:
Improve error checking and exception handling for date and time

vivienherq committed:
Merge pull request #13 from vivienherq/branch-A-MoreErrorHandling

Branch A-MoreErrorHandling

vivienherq committed:
Add delay before exiting and  closing GUI

vivienherq committed:
Add personality for Dude

vivienherq committed:
Merge pull request #14 from vivienherq/branch-A-Personality

Branch A-Personality

vivienherq committed:
Improve exception handling for parser

vivienherq committed:
Upload image of UI

vivienherq committed:
Improve code quality

vivienherq committed:
Update README.md

vivienherq committed:
Find command: support searching notes

vivienherq committed:
Update README.md

vivienherq committed:
Refactor classes into packages

vivienherq committed:
Configure build.gradle to create fat JAR file
